pkg/api/v1/defaults.go

```diff
@@ -160,6 +160,18 @@ func SetDefaults_Pod(obj *Pod) {
 			}
 		}
 	}
+	for i := range obj.Spec.InitContainers {
+		if obj.Spec.InitContainers[i].Resources.Limits != nil {
```

Comment:
there are some proposals to skip defaulting altogether in pod specs inside objects like that, and keep the content of a deployment or replicaset limited to exactly what the user set